plugins/modules/folder.py

```diff
@@ -0,0 +1,292 @@
+            - The folder type controls what resources can be held in the folder, as well as the path where the folder is created.
+            - For example, a folder at path /DC-01/vm/my/folder has folder type 'vm'.
+            - Only used if the O(relative_path) option is used.
+        type: str
+        required: false
+        choices: [vm, host, storage, network]
+    relative_path:
+        description:
+            - The relative path of the folder to create. The relative path should not include the datacenter or the folder type.
+            - For example the relative path for the folder /DC-01/vm/my/folder is my/folder
+            - One of O(relative_path) or O(absolute_path) must be specified.
+        type: str
+    absolute_path:
+        description:
+            - The absolute path of the folder to create. The absolute path should include the datacenter or the folder type.
+            - The leading slash is not required. For example the absolute path could be /DC-01/vm/my/folder or DC-01/vm/my/folder
+            - One of O(relative_path) or O(absolute_path) must be specified.
+        type: str
+    state:
+        description:
+            - Create V(present) or remove V(absent) a vSphere folder.
+            - When state is V(present), all folders in the specified path will be created if they are missing.
+            - When state is V(absent), all content inside the folder will be deleted. See O(remove_vm_data)
+        choices: [ absent, present ]
+        default: present
+        type: str
+    remove_vm_data:
+        description:
+            - Only used if state is V(absent)
+            - If true, any VMs in the folder tree will be completely removed. This includes any disks or data associated with the VM.
+            - >-
+              If false, any VMs in the folder tree are deregistered. This means they are removed from the vSphere inventory but
+              their data is not deleted.
+        default: false
+        type: bool
+
+extends_documentation_fragment:
+    - vmware.vmware.base_options
+'''
+
+EXAMPLES = r'''
+- name: Create Folder Using A Relative Path (/DC01/storage/my/test/folder)
+  vmware.vmware.folder:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    validate_certs: false
+    datacenter: DC01
+    relative_path: my/test/folder
+    folder_type: storage
+
+# This is the same as the example above, but shows how to do it using different params.
+- name: Create A Folder Using An Absolute Path
+  vmware.vmware.folder:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    validate_certs: false
+    absolute_path: /DC01/storage/my/test/folder
+
+
+- name: Delete The Folder and All Of Its Contents
+  vmware.vmware.folder:
+    hostname: "https://vcenter"
+    username: "username"
+    password: "password"
+    validate_certs: false
+    absolute_path: /DC01/storage/my
+    state: absent
+
```

Comment:
By default `remove_vm_data` is false
```suggestion
remove_vm_data: true
```